src/Playwright.Tests/ConventionTests.cs

```diff
@@ -0,0 +1,32 @@
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Playwright.Tests;
+
+public class ConventionTests
+{
+    // To ensure that public method are not inlined by the new tiered PGO JIT mode,
+    // we need to mark them with [MethodImpl(MethodImplOptions.NoInlining)]
+    [Test]
+    public void EnsurePublicMethodsAreNotInlined()
+    {
+        var assembly = typeof(Playwright).Assembly;
+
+        var types = assembly.GetTypes()
+            .Where(t => t.IsClass && !t.IsAbstract)
+            .Where(t => t.BaseType != null && t.BaseType.Name == "ChannelOwnerBase");
+
+        foreach (var type in types)
+        {
+            var methods = type
+                .GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
+                .Where(m => !m.IsSpecialName)
+                .Where(m => m.Name != "Dispose" && m.Name != "DisposeAsync" && m.Name != "ToString");
+
+            foreach (var method in methods)
+            {
+                Assert.True(method.MethodImplementationFlags.HasFlag(MethodImplAttributes.NoInlining), $"{type.Name}.{method.Name} is not marked with [MethodImpl(MethodImplOptions.NoInlining)]");
```

Comment:
If you put these messages into a list and then assert the list is empty, you could fail the test with everything that needs to be fixed at once in a single message that concatenates them.

Reply:
Sure, tell me now after I reran it 50 times!